frontend/src/logger/pcm.logger.ts

```diff
@@ -0,0 +1,75 @@
+import {AxiosInstance} from 'axios'
+import {AppConfig} from '../app-config/types'
+import {handleNotAuthorizedErrors} from '../auth/handleNotAuthorizedErrors'
+import identityFn from 'lodash/identity'
+
+enum LogLevel {
+  info = 'INFO',
+  warning = 'WARNING',
+  debug = 'DEBUG',
+  error = 'ERROR',
+  critical = 'CRITICAL',
+}
+
+interface LogEntry {
+  message: string
+  level: LogLevel
+  extra?: {[key: string]: string}
```

Comment:
issue(non-blocking): if we plan to support nested objects in the `extra` parameter, a better type signature could be `Record<string, unknown>`